src/rust/engine/src/nodes.rs

```diff
@@ -1285,19 +1343,21 @@ pub enum NodeOutput {
   DirectoryListing(Arc<DirectoryListing>),
   LinkDest(LinkDest),
   ProcessResult(Box<ProcessResult>),
-  Snapshot(Arc<store::Snapshot>),
```

Comment:
We no longer use `NodeOutput::Snapshot`.

Reply:
Interesting.

So, FWIW: a (minor) consideration here is that both the `NodeKey` and the `NodeOutput` are kept in memory. Smaller inputs/outputs are better for memory usage (possibly also for performance if they're being copied), but might mean more work to get the value that will actually be used by a consumer.

As an example: not holding `Snapshot` in memory is a win from a memory usage perspective, because we will load the `Snapshot` from the LMDB store any time is used. That might take some time, but is very minor relative to re-capturing the `Snapshot`. On the other hand, `FileListing` is now being kept in memory, and I don't really see an alternative since there is no more useful key to identify it with (i.e. no equivalent of `Digest` for this case).

...not going anywhere in particular with this. Lots of angles, and unclear what is faster or slower without benchmarking. And very likely that lots of these things would not make a difference.